internal/store/testing_store.go

```diff
@@ -79,8 +91,43 @@ func (s *TestingStore) Actions() []Action {
 	return append([]Action{}, s.actions...)
 }
 
+func (s *TestingStore) ProcessActions(ctx context.Context, reducer Reducer, subscribers []Subscriber) {
```

Comment:
have you considered having a custom TestStore that implements a custom Dispatch function like here?
https://github.com/tilt-dev/tilt/blob/master/internal/engine/local/controller_test.go#L174

Honestly, this function has a lot of footguns that I can see off the bat. I'm not confident I would be able to use it correctly.

Reply:
I went down that road originally and hit a deadlock since the subscriber dispatches (while holding the lock), and once you start making a mock `Dispatch` look like async (and potentially want to get subscribers involved as here), it starts looking like a weird version of the real store.

I think the best course is actually to do this in 2 parts - the tests don't _really_ care about actions (nor should they TBH), so I'll refactor them to just poll on state, and then this PR can use the real store with a minimal reducer.

Reply:
I did see it and play with it some - with the way the tests like to introspect the actions as they go, it was making the race detector go crazy (which I've been relying on a fair amount for these tests). It's not an unresolvable problem but would have meant largely rewriting the tests anyway.

Take a look at #4296 - I think it's honestly a better approach for where we're going as it should make updating these tests for API server very straightforward as it can just use a fake client instead of store/state. I can rebase this on those changes.

Reply:
Okay, this PR is updated to not touch `TestingStore` and the tests look 90% the same as they did before 🎉 

(I'm still doing other cleanup around relative path fixes I made so leaving in draft)